backends/core/models.py

```diff
@@ -249,9 +316,48 @@ def destroy(self):
       Param.destroy(*param_ids)
     self.delete()
 
+  def get_status(self, max_retires=10):
+    key = str(self.pipeline_id) + '_' + str(self.id) + '_' + CACHE_KEY_STATUS
+    retries = 0
+    while retries < max_retires:
+      status = self.memcache_client.gets(key)
+      if status is None: 
+        '''The status is uninitialized in memcache, 
+          getting it from the database and
+          updating its value in memcache
+        '''
+        self.memcache_client.set(key, self.status, time=MEMCACHE_DEFAULT_EXPIRATION_TIME_SECONDS)
+        return self.status
+      if self.memcache_client.cas(key, status):
+        return status
+      retries += 1
+
+  def prepare_for_start(self, max_retries=10):
+    '''
```

Comment:
For multiline strings you can safely use the triple double quote `"""`. I am realizing how it is weird :) but it's the standard.

Reply:
done